### PR TITLE
Gs profiles

### PR DESCRIPTION
**Pull request by gavinschriver**
# Description
-add conditional interface  for promoting/demoting user roles from ProfileList view
## Type of change
- [ ] New feature (non-breaking change which adds functionality)
# How Has This Been Tested?
Please describe step-by-step the process that a reviewer can follow to adequately test this PR. 
- [ ] make sure you have the most recent sever main
- [ ] log in as me@me.me with password me and navigate to Profiles view, make sure you see user names but no options
- [ ] log out and back in as admin with password admin, navigate to profiles, make sure you can see two users. toggle me@me.me to Admin, then log out and back in as Me and make sure you can see those same controls. try to make yourself an author and make sure you get a warning prompt. close out, make sure everything is as before, then do the same but follow thru with the change and make sure the page rerenders without access to those admin controls.
# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings

### CODE REVIEW

File: src/components/profiles/ProfileProvider.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(isStaffUpdate),
+    }).then(getProfiles);
+  };
+
+  return (
+    <ProfileContext.Provider
+      value={{
+        ProfileContext,
```
Comment (skratz17) [inline src/components/profiles/ProfileProvider.js:32]: It probably doesn't matter but I guess the `ProfileContext` does not need to be passed in as a value here?

File: src/components/profiles/ProfileStatusToggle.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      const isStaffValue = e.target.value;
+      updateUserRole(userId, { is_staff: isStaffValue }).then(() =>
+        setIsSubmitting(false)
+      );
+    }
+  };
+
+  //runs when a user (admin) confirms self-demotion; re-pushing the path to the profiles list is necessary to re-render
+  // after removing the "is_admin" token
+  const handleSelfNuke = (e) => {
```
Comment (skratz17) [inline src/components/profiles/ProfileStatusToggle.js:31]: function name is metal as hell

Comment (skratz17) [inline src/components/profiles/ProfileStatusToggle.js:31]: also what an excellent touch, this whole prompt the user they're gonna nuke themselves and require confirmation AND destroy the localStorage token and redirect them so their view updates all so good

File: src/components/profiles/ProfileProvider.js (hunk 1 above)
Comment (gavinschriver) [reply src/components/profiles/ProfileProvider.js:32]: guuudd call i think that was an autofill maybe or i was just gettin strange

**Review by skratz17 — APPROVED** [review]

**Review by Dylan-Trevor-Rowe — APPROVED** [review]
This works swell

**Pull request merged**